game.js

```diff
@@ -1,12 +1,179 @@
-var a = 0
-document.getElementById('button').onclick = Click
-function Click() {
-  document.getElementById('button').innerHTML = a
-  a = a + 1
-  for (var i = 0; i < a; i++){
-    var li = document.createElement("LI");
-    var textnode = document.createTextNode("Cica");
-    li.appendChild(textnode)
-    document.getElementById("myList").appendChild(li);
+// GLOBALS
+// GLOBALS of GRID.
+maxColumns = 10;
+maxRows = 10;
+startDrawX = 150;
+startDrawY = 300;
+var tileArray = [];
+
+// GLOBALS of BALANCE.
+var trashMaximum = 10000; // Sets the amount of trash to distribute in the game.
+var trashMinimum = 0; // The minimum amount of trash on a tile.
+var trashVariance = 1500; // The maximum that can be added to a tile at random for higher variance.
+var trashDistributionPower = 1.25; // Lower values will result in more evenly distributed trash. Higher values push trash to the edges of the map.
+var startingRow = 3 + Math.floor(Math.random()*(maxRows-3)); // Starting square cannot be column 0, 1, 2 or column 7, 8, 9.
+var startingColumn = 3 + Math.floor(Math.random()*(maxColumns-3)); // Starting row cannot be row 0, 1, 2 or column 7, 8, 9.
+var startingClearCells = 4; // Clear this many cells INCLUDING the starting cell.
+
+// CLLASSES
+class InGameTile {
+  // InGametile points to an object that is drawn and updated on events.
+  // InGametile also holds some information about itself.
+  constructor(whichRow, whichColumn) {
+    // TILE DECLEARATION
+    this.tileIndex = 0; // Waits for a function declaration. Stores the index of the tile inside the tileArray.
+    this.row = whichRow; // Tells you which row this grid element sits in.
+    this.column = whichColumn; // Tells you which column this grid element sits in.
+    this.trashAmount = trashMaximum; // Amount of trash on tile.
+
+    // Creates a grid element for the corresponding tile.
+    this.gridRepresentator = document.createElement("DIV");
+    grid.appendChild(this.gridRepresentator);
+    this.gridRepresentator.innerHTML = "";
+    this.gridRepresentator.className = "tile"; // Formats the InGameTile object per style.css class "tile".
+    this.gridRepresentator.style.left = locX + "px"; // Locks object at provided X.
+    this.gridRepresentator.style.top = locY + "px"; // Locks object at provided Y.
   }
+
+  addHouse() {
+    var img = document.createElement("img");
+    img.src = "images/tree.png";
+    this.gridRepresentator.appendChild(img);
+  }
+
+}
+
+// FLOURISH / COSMETIC FUNCTIONS
+// Without which the game will run but will heavily impact user experience.
+function gridEl_updateTrash(whichTile) {
+  // Change the trash display. This is intended to be temporary.
+  if (whichTile.trashAmount !== 0) {
+    whichTile.gridRepresentator.innerHTML = Math.floor(whichTile.trashAmount);
+  } else {
+    // Should later call the clear function.
+    whichTile.gridRepresentator.innerHTML = "";
+  }
+
+  // Color the grid based on how full of trash it is.
+  var tempReal = 0;
```

Comment:
az ilyen kosza globalis valtozok nem tul szerencsesek. lehet, h te nem akartad globalisnak, de ha nem egy functionon vagy egy objecten belul van definialva, akkor minden valtozo globalis javascriptben, es meghozza mindegyik mar a dokumentum elejetol ervenyes, vagyis ez a valtozo mar a 0. soron is tudja, h az o erteke 0 hiaba csak az 58. soron definialtad. szoval:
- ha nem akarod globalisnak marpedig, ha nem kell, h globalis legyen, akkor ne legyen az. erdemes a globalis valtozok szamat minimalison tartani. na szoval ha nem akarod globalisnak, akkor tedd bele egy functionbe
- ha globalisnak szeretned akkor deklarald a doksi elejen, h egybol vilagos legyen milyen valtozok vannak ebben a dokumentumban.